src/ProjectTask.php

```diff
@@ -1802,6 +1808,30 @@ public static function recalculatePercentDone($ID)
         return true;
     }
 
+    /**
+     * Recalculate the status of a project task based on the percent_done.
+     * @since 11.0.0
+     * @param array $input
+     */
+    public static function recalculateStatus(array $input)
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        if (!$input['auto_projectstates']) {
+            return false;
+        }
+        $config = \Config::getConfigurationValues('core');
+        if ($input['percent_done'] === 0) {
+            $state_id = $config['projecttask_unstarted'] ?? 0;
+        } elseif ($input['percent_done'] === 100) {
+            $state_id = $config['projecttask_completed'] ?? 0;
+        } else {
+            $state_id = $config['projecttask_inprogress'] ?? 0;
+        }
+        return $state_id;
```

Comment:
```suggestion
        $state = ProjectState::getById($state_id):
        if (!$state) {
            return false;
        }
        return $state->getID();
```

Maybe we can verify that the state exists (it might have been deleted).